grants_tagger_light/preprocessing/preprocess_mesh.py

```diff
@@ -0,0 +1,172 @@
+    data_batch = []
+    for item in tqdm(yield_raw_data(input_path), total=15_000_000):  # approx 15M docs
+        processed_item = process_data(item, filter_tags, filter_years)
+
+        if processed_item:
+            data_batch.append(processed_item)
+
+        if len(data_batch) >= buffer_size:
+            yield data_batch
+            data_batch = []
+
+    if data_batch:
+        yield data_batch
+
+
+def preprocess_mesh(
+    raw_data_path,
+    processed_data_path,
+    mesh_tags_path=None,
+    filter_years=None,
+    n_max=None,
+    buffer_size=10_000,
+):
+    if mesh_tags_path:
+        filter_tags_data = pd.read_csv(mesh_tags_path)
+        filter_tags = filter_tags_data["DescriptorName"].tolist()
+        filter_tags = set(filter_tags)
+    else:
+        filter_tags = None
+
+    if filter_years:
+        min_year, max_year = filter_years.split(",")
+        filter_years = [int(min_year), int(max_year)]
+
+    # If only using a tiny set of data, need to reduce buffer size
+    if n_max is not None and n_max < buffer_size:
+        buffer_size = n_max
+
+    with open(processed_data_path, "w") as f:
+        for i, data_batch in enumerate(
+            yield_data(
+                raw_data_path, filter_tags, filter_years, buffer_size=buffer_size
+            )
+        ):
+            write_jsonl(f, data_batch)
+            if n_max and (i + 1) * buffer_size >= n_max:
+                break
+
+
+preprocess_mesh_app = typer.Typer()
+
+
+@preprocess_mesh_app.command()
+def preprocess_mesh_cli(
+    input_path: Optional[str] = typer.Argument(None, help="path to BioASQ JSON data"),
+    train_output_path: Optional[str] = typer.Argument(
+        None, help="path to JSONL output file that will be generated for the train set"
+    ),
+    label_binarizer_path: Optional[Path] = typer.Argument(
+        None, help="path to pickle file that will contain the label binarizer"
+    ),
+    test_output_path: Optional[str] = typer.Option(
+        None, help="path to JSONL output file that will be generated for the test set"
+    ),
+    mesh_tags_path: Optional[str] = typer.Option(
+        None, help="path to mesh tags to filter"
+    ),
+    test_split: Optional[float] = typer.Option(
+        0.01, help="split percentage for test data. if None no split."
+    ),
+    filter_years: Optional[str] = typer.Option(
+        None, help="years to keep in form min_year,max_year with both inclusive"
+    ),
+    n_max: Optional[int] = typer.Option(
+        None,
+        help="""
+        Maximum limit on the number of datapoints in the set
+        (including training and test)""",
+    ),
+):
+    params_path = os.path.join(os.path.dirname(__file__), "../params.yaml")
+    with open(params_path) as f:
+        params = yaml.safe_load(f)
+
+    # Default values from params
+    if not mesh_tags_path:
+        mesh_tags_path = params["preprocess_bioasq_mesh"].get("mesh_tags_path")
+
+    # Default value from params
+    if not filter_years:
```

Comment:
the logic until this point was inserted so a user could overide the default params. i think this is not a good idea anymore. we should not overide the params and dvc should use the params when it calls the command. i understand this might be unclear, let me know if you need clarification

Reply:
So the params `mesh_tags_path` and `filter_years` that are pased to the function should not be used anymore right? Because they would be directly taken from the `params` argument

Reply:
The opposite, you should remove the overwrite logic i.e. params and only allow for the params to be taken from what is being passed into the function